src/db.c

```diff
@@ -1021,8 +1152,12 @@ void scanGenericCommand(client *c, robj *o, unsigned long cursor) {
 
     /* Handle the case of a hash table. */
     ht = NULL;
+    /* During main dictionary traversal in cluster mode, 48 lower bits in the cursor are used for positioning in the HT.
+     * Following 14 bits are used for the slot number, ranging from 0 to 2^14-1.
+     * Slot is always 0 at the start of iteration and can be incremented only in cluster mode. */
+    int slot = getAndClearSlotIdFromCursor(&cursor);
```

Comment:
It struck me that this code applies runs not only to SCAN but also on HSCAN, SSCAN, etc. as long as it's implemented as a dict. We should check `o == NULL` to make sure it applies only to SCAN.

```suggestion
    int slot = (o == NULL) ? getAndClearSlotIdFromCursor(&cursor) : 0;
```